src/map/entities/charentity.cpp

```diff
@@ -1509,6 +1520,11 @@ void CCharEntity::OnItemFinish(CItemState& state, action_t& action)
 
 CBattleEntity* CCharEntity::IsValidTarget(uint16 targid, uint16 validTargetFlags, std::unique_ptr<CMessageBasicPacket>& errMsg)
 {
+    // check /blockaid
+    CCharEntity* PChar = (CCharEntity*)this->GetEntity(targid, TYPE_PC);
+    if (PChar && charutils::IsAidBlocked(this, PChar))
+        return nullptr;
```

Comment:
is there no specialized error message for trying to cast on someone with blockaid?